src/osd/scrubber/scrub_reservations.cc

```diff
@@ -194,20 +230,30 @@ void ReplicaReservations::verify_rejections_source(
 		  *op->get_req())
 	   << dendl;
 
+  // is this a stale response to a previous request (e.g. one that
+  // timed-out)? if so - ignore it (but do log an error, to cause tests
+  // to fail)
+  const auto msg_epoch = op->get_req<MOSDScrubReserve>()->map_epoch;
+  const auto msg_nonce = op->get_req<MOSDScrubReserve>()->reservation_nonce;
+  const auto nonce_verified = is_response_relevant(msg_epoch, msg_nonce, from);
+  if (!nonce_verified) {
+    m_osds->clog->warn() << nonce_verified.error();
```

Comment:
This shouldn't be a warning -- it's a normal if rare component of the protocol.  If the cancellation can't order past the original request, I think it'll actually happen every time.

Reply:
Good point.
I'll remove the warning. The idea was to notice such events, to detect possible bugs that would cause this scenario other than the known scenario of a very very delayed connection.